docs/faq/ImmutableData.md

```diff
@@ -0,0 +1,245 @@
+# Redux FAQ: Immutable Data
+
+## Table of Contents
+- [What are the benefits of Immutability?](#benefits-of-immutability)
+- [Why is immutability required in Redux?](#why-is-immutability-required)
+- [Do I have to use Immutable.JS?](#do-i-have-to-use-immutable-js)
+- [What are the issues with using ES6 for immutable operations?](#issues-with-es6-for-immutable-ops)
+- [Why should I use an immutable-focused library such as Immutable.JS?](#why-use-immutable-library)
+- [Why should I choose Immutable.JS as an immutable library?](#why-choose-immutable-js)
+- [What are the issues with using Immutable.JS?](#issues-with-immutable-js)
+- [Is Immutable.JS worth the effort?](#is-immutable-js-worth-effort)
+- [What are the Recommended Best Practices for using Immutable.JS with Redux?](#immutable-js-best-practices)
+
+<a id="benefits-of-immutability"></a>
+## What are the benefits of immutability?
+
+Immutability can bring increased performance to your app, and leads to simpler programming and debugging, as data that never changes is easier to reason about than data that is free to be changed arbitrarily throughout your app.
+
+In particular, immutability in the context of a Web app enables sophisticated change detection techniques to be implemented simply and cheaply, ensuring the computationally expensive process of updating the DOM occurs only when it absolutely has to (a cornerstone of React’s performance improvements over other libraries).
+<a id="why-is-immutability-required"></a>
+## Why is immutability required in Redux?
+
+In the context of Redux, if your reducers mutate their arguments and directly modify the state tree, you will cause React Redux’s `connect` function to break, as it will be unable to detect changes made to objects and arrays. You MUST therefore use immutable operations in your reducers when using Redux.
+<a id="do-i-have-to-use-immutable-js"></a>
+## Do I have to use Immutable.JS?
```

Comment:
> First, I think I'd like to see the question rephrased/expanded as "What approaches are there for handling data immutably? Do I have to use Immutable.js?".

I've rephrased the question, but not answered the expanded part (i.e. 'What approaches are there for handling data immutably?'). I'll do that in another PR.

> Second, I would like to see a mention of three possible approaches: "plain JS", immutable update libraries, and Immutable.js, with some relatively short lists of tradeoffs (both pro and con) for each.

Again, I'll do that in a separate PR.

> Third, I'd prefer to refer to "plain" Javascript rather than ES6 specifically, since the object spread operator isn't even final yet (so ES2017 at earliest), and while Object.assign is ES6, the same capability exists with _.extend and its equivalents.

Done.